server/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalNumericMetricsAggregation.java

```diff
@@ -75,11 +75,11 @@ protected MultiValue(StreamInput in) throws IOException {
             super(in);
         }
 
+        @Override
         public abstract double value(String name);
 
-        public String valueAsString(String name) {
-            return format.format(value(name)).toString();
-        }
+        @Override
+        public abstract String valueAsString(String name);
```

Comment:
this change wasn't strictly necessary, however it seems cleaner to me to _force_ every implementation to think about it.

The alternative is to keep the old version, but only overwrite it in TopMetrics

Reply:
Could you link me to what that'd look like? Maybe make the change on top of this PR and push it to another branch or something? Just somewhere where I can see it. I think maybe we should stop `top_metrics` from extending `InternalNumericMetricsAggregation.MultiValue`. Like you said somewhere else - it ain't really numeric any more.

Reply:
I marked it in the PR, fixing `top_metrics` to not extend `InternalNumericMetricsAggregation.MultiValue` would be the alternative of course. However sounds like a larger re-factoring to me. If that's what you are aiming for, I suggest to revert the above and go with the smaller change for now.